src/Microsoft.AspNet.Mvc.Razor/RazorView.cs

```diff
@@ -179,7 +179,7 @@ private bool EnableInstrumentation
                     // the body content. Throwing this exception wouldn't return a 500 (since content has already been
                     // written), but a diagnostic component should be able to capture it.
 
-                    var message = Resources.FormatLayoutCannotBeRendered("FlushAsync");
+                    var message = Resources.FormatLayoutCannotBeRendered(Path, "FlushAsync");
```

Comment:
the issue is the `RazorPage` property hiding the class.  but `nameof(Razor.RazorPage.FlushAsync)` works.